src/aiori-HDF5.c

```diff
@@ -634,12 +931,12 @@ static void SetupDataSet(void *fd, IOR_param_t * param)
                 WARN("unable to determine HDF5 version for 'no fill' usage");
 #endif
                 dataSet =
-                    H5Dcreate(*(hid_t *) fd, dataSetName, H5T_NATIVE_LLONG,
-                              dataSpace, dataSetPropList);
+                    H5Dcreate2(*(hid_t *) fd, dataSetName, H5T_NATIVE_LLONG,
```

Comment:
some of these generic changes should be pushed upstream to the ior repo.
the ones that have nothing to do with fault injection.

Reply:
Changed.